packages/components/src/components/hds/app-header/home-link.ts

```diff
@@ -12,21 +12,40 @@ import type { HdsInteractiveSignature } from '../interactive/';
 export interface HdsAppHeaderHomeLinkSignature {
   Args: HdsInteractiveSignature['Args'] & {
     icon: HdsIconSignature['Args']['name'];
+    isIconOnly?: boolean;
     color?: string;
-    ariaLabel: string;
+    text: string;
   };
   Element: HdsInteractiveSignature['Element'];
 }
 
 export default class HdsAppHeaderHomeLink extends Component<HdsAppHeaderHomeLinkSignature> {
-  get ariaLabel(): string {
-    const { ariaLabel } = this.args;
+  get text(): string {
+    const { text } = this.args;
 
     assert(
-      '@ariaLabel for "Hds::AppHeader::HomeLink" ("Logo") must have a valid value',
-      ariaLabel !== undefined
+      '@text for "Hds::AppHeader::HomeLink" must have a valid value',
+      text !== undefined
     );
 
-    return ariaLabel;
+    return text;
+  }
+
+  get icon(): HdsIconSignature['Args']['name'] {
+    const { icon } = this.args;
+
+    assert(
+      '@icon for "Hds::AppHeader::HomeLink" must have a valid value',
+      icon !== undefined
+    );
+
+    return icon;
+  }
+
+  get isIconOnly(): boolean {
+    if (this.icon) {
+      return this.args.isIconOnly ?? false;
```

Comment:
[question] why is this set `false` by default, when it's the exception? practically everywhere will be icon-only so consumers will have to explicitly declare it with this implementation (a bit of anti-pattern for us)

Reply:
Good point, I've simplified the logic a bit based on @KristinLBradley 's suggestion and set this to true be default.

Reply:
That's a good point. @jorytindall try changing `false` to `true` here. Also, I think you can simplify the logic as I suggested before: https://github.com/hashicorp/design-system/pull/2951/files#r2159438378